sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/PreferMethodReferenceCheck.java

```diff
@@ -424,17 +424,22 @@ private static boolean isAnonClass(DetailAST ast) {
      *     {@code false} otherwise
      */
     private static boolean isArgUsedInArrayInit(DetailAST declarator, String name) {
-        final DetailAST nestedDecl = declarator.findFirstToken(TokenTypes.ARRAY_DECLARATOR);
-        final DetailAST expr = declarator.findFirstToken(TokenTypes.EXPR);
-        boolean result = false;
-        if (nestedDecl == null && expr != null) {
-            final DetailAST ident = expr.findFirstToken(TokenTypes.IDENT);
-            result = ident != null && name.equals(ident.getText());
-        }
-        else if (nestedDecl != null && expr == null) {
-            result = isArgUsedInArrayInit(nestedDecl, name);
```

Comment:
No need for recursion here anymore, we can just iterate through the sequential `ARRAY_DECLARATOR` brackets and check if they have an `EXPR` child (i.e. `[2]`), then simply check the first `[a]` to make sure that the lambda parameter and expression match.